mrmustard/lab_dev/transformations/czgate.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The class representing a controlled-phase gate.
+"""
+
+from __future__ import annotations
+
+from typing import Sequence
+from mrmustard import math
+from mrmustard.physics.ansatz import PolyExpAnsatz
+
+from .base import Unitary
+from ..utils import make_parameter
+
+__all__ = ["CZgate"]
+
+
+class CZgate(Unitary):
+    r"""Controlled Z gate.
+
+    It applies to a single pair of modes. One can optionally set bounds for each parameter, which
+    the optimizer will respect.
+
+    .. math::
+
+        C_Z = \exp(ig q_1 \otimes q_2 / \hbar).
```

Comment:
is g the same as s? should we use the same letter if they are the same?

Reply:
Yes, they are the same thing. I will make this change